transformations/whitespace_perturbation/transformation.py

```diff
@@ -1,33 +1,37 @@
+import itertools
 import random
 
 from interfaces.SentenceOperation import SentenceOperation
 from tasks.TaskTypes import TaskType
 
 
-def whitespace(text, remove_prob=0.1, add_prob=0.05, seed=0):
+def whitespace(text, remove_prob=0.1, add_prob=0.05, seed=0, max_outputs=1):
     random.seed(seed)
-    newtext = ''
-    for char in text:
-        random_num = random.random()
-        if char.isspace() and random_num < remove_prob:
-            continue
-        newtext += char
-        if (not char.isspace()) and random_num < add_prob:
-            newtext += ' '
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
```

Comment:
Why not just use `range(max_outputs)`?

Reply:
I agree that `range(max_outputs`) is better and changed it in the latest commit. To answer your question, this particular line of code is actually from the example task [Butter Fingers Perturbation](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/transformations/butter_fingers_perturbation/transformation.py).